client/app/bundles/course/users/components/buttons/PointManagementButtons.tsx

```diff
@@ -48,20 +50,29 @@ const translations = defineMessages({
 });
 
 const PointManagementButtons: FC<Props> = (props) => {
-  const { intl, permissions, data, isDirty, isManuallyAwarded, handleSave } =
-    props;
+  const {
+    intl,
+    permissions,
+    data,
+    isDirty,
+    isManuallyAwarded,
+    handleSave,
+    studentId,
+    isErrorInput,
+    isDownloading,
+  } = props;
   const dispatch = useAppDispatch();
   const [isSaving, setIsSaving] = useState(false);
   const [isDeleting, setIsDeleting] = useState(false);
 
   const onSave = (): void => {
     setIsSaving(true);
-    dispatch(updateExperiencePointsRecord(data))
-      .then((response) => {
+    dispatch(updateExperiencePointsRecord(data, studentId))
+      .then(() => {
```

Comment:
Why is this removed?

Reply:
because it's not needed anymore in the subsequent operation..

Reply:
Why do we not use the backend response data in updating the state?

Reply:
this can be done. I have refactored properly